File: go-base-web/src/api/productType.ts
import request from '@/utils/request'
import type { ApiResponse, PageResponse } from '@/types'
import type { ProductType, CreateProductTypeRequest, UpdateProductTypeRequest, ProductTypeQuery, OptionItem } from '@/types/productType'

// 获取产品类型列表
export function getProductTypeList(params: ProductTypeQuery) {
  return request.get<any, ApiResponse<PageResponse<ProductType>>>('/productType', { params })
}

// 获取产品类型详情
export function getProductType(id: number) {
  return request.get<any, ApiResponse<ProductType>>(`/productType/${id}`)
}

// 创建产品类型
export function createProductType(data: CreateProductTypeRequest) {
  return request.post<any, ApiResponse<ProductType>>('/productType', data)
}

// 更新产品类型
export function updateProductType(id: number, data: UpdateProductTypeRequest) {
  return request.put<any, ApiResponse<ProductType>>(`/productType/${id}`, data)
}

// 删除产品类型
export function deleteProductType(id: number) {
  return request.delete<any, ApiResponse>(`/productType/${id}`)
}

// 批量删除产品类型
export function batchDeleteProductType(ids: number[]) {
  return request.delete<any, ApiResponse>('/productType/batch', { data: { ids } })
}

// 获取产品类型选项列表
export function getProductTypeOptions(params?: { display_field?: string; count_table?: string; count_field?: string; exclude_deleted?: boolean; count_created_by?: number }) {
  return request.get<any, ApiResponse<OptionItem[]>>('/productType/options', { params })
}

File: go-base-server/router/modules/productType.go
package modules

import (
	"github.com/gin-gonic/gin"

	v1 "go-base-server/api/v1"
	"go-base-server/model/request"
	"go-base-server/router/registry"
)

func init() {
	RegisterModule(&ProductTypeModule{})
}

type ProductTypeModule struct{}

func (m *ProductTypeModule) Name() string {
	return "产品类型"
}

func (m *ProductTypeModule) RegisterPublicRoutes(rg *gin.RouterGroup) {
	// 无公开路由
}

func (m *ProductTypeModule) RegisterPrivateRoutes(rg *gin.RouterGroup) {
	R(rg, "GET", "/productType", m.Name(), "产品类型列表", v1.ProductType.GetProductTypeList,
		registry.WithAuth(), registry.WithRequest(request.ProductTypeListRequest{}))
	R(rg, "GET", "/productType/options", m.Name(), "产品类型选项", v1.ProductType.GetProductTypeOptions, registry.WithAuth())
	R(rg, "GET", "/productType/:id", m.Name(), "产品类型详情", v1.ProductType.GetProductType, registry.WithAuth())
	R(rg, "POST", "/productType", m.Name(), "创建产品类型", v1.ProductType.CreateProductType,
		registry.WithAuth(), registry.WithRequest(request.CreateProductTypeRequest{}))
	R(rg, "PUT", "/productType/:id", m.Name(), "更新产品类型", v1.ProductType.UpdateProductType,
		registry.WithAuth(), registry.WithRequest(request.UpdateProductTypeRequest{}))
	R(rg, "DELETE", "/productType/:id", m.Name(), "删除产品类型", v1.ProductType.DeleteProductType, registry.WithAuth())
	R(rg, "DELETE", "/productType/batch", m.Name(), "批量删除产品类型", v1.ProductType.BatchDeleteProductType,
		registry.WithAuth(), registry.WithRequest(request.BatchDeleteProductTypeRequest{}))
	// 导入导出
	R(rg, "GET", "/productType/export", m.Name(), "导出产品类型", v1.ProductType.ExportProductType, registry.WithAuth())
	R(rg, "POST", "/productType/import", m.Name(), "导入产品类型", v1.ProductType.ImportProductType, registry.WithAuth())
	R(rg, "GET", "/productType/template", m.Name(), "下载导入模板", v1.ProductType.DownloadTemplateProductType, registry.WithAuth())
}
